torch/fx/passes/_tensorify_python_scalars.py

```diff
@@ -281,6 +306,27 @@ def _sympy_interp(expr: sympy.Expr) -> MetaProxy:
                     node.replace_all_uses_with(guard_scalar(val))
                     graph.erase_node(node)
 
+    # Sometimes by the time we get to tensorify, there have already been
+    # specializations, eg. in python_arg_parser.h. In these cases,
+    # placeholder nodes no longer have a reference to their original
+    # symfloat and thus we need to deduce specializations have happend
+    # via shape_env.replacements. NB: there's an important invariant here
+    # that symfloats keep consistent names across restarts.
+    for k, v in shape_env.var_to_val.items():
+        if symbol_is_type(k, SymT.FLOAT) and isinstance(v, sympy.core.numbers.Float):
+            name = str(k)
+            if (
+                not TensorifyState.should_specialize(name)
+                and name not in tensorified_symbols
+            ):
+                TensorifyState.specialize(name)
+                should_restart = True
+
+    if should_restart:
+        # Sledgehammer time. Restart dynamo analysis, keeping track of which input sources
+        # are no longer needed and should be specialized.
+        raise TensorifyScalarRestartAnalysis
```

Comment:
Similar to https://github.com/pytorch/pytorch/pull/131983 need to clear speculation log

Reply:
Not sure what happens if compiler collectives is going on at this point too

Reply:
I'm looking for a comment that says something like "restarting analysis is necessary because we need to instruct Dynamo to NOT make these as inputs."

It really does seem like CUDA graph should be able to ignore inputs that don't get used inside the body though LOL